ConwaysGameOfLife.Tests/GameControllerShould.cs

```diff
@@ -9,64 +10,121 @@ public class GameControllerShould
+        [InlineData(new string[] { ",," }, "10,10")]
+        [InlineData(new string[] { "3,3," }, "10,10")]
+        [InlineData(new string[] { "3,3,3", "3;3;" }, "3,3")]
+        [InlineData(new string[] { "a,b" }, "3,3")]
+        [InlineData(new string[] { "-1,4", "4,-1" }, "3,3")]
+        [InlineData(new string[] { "10,0", "0,10" }, "3,3")]
+        public void ReturnUniverseString_GivenInvalidInputsFollowedByValidInput(IEnumerable<string> invalidInputs, string validInput)
+        {
+            _input.GetReadLine(invalidInputs);
+            _input.GetReadLine(validInput);
+            var expectedOutputLine1 = "Invalid input. Please try again.";
+            var expectedOutputLine2 = "Please enter the width & length for the Universe as a number followed by a comma then a number (e.g. '0,0').";
+
+            var result = _controller.CreateValidUniverseString();
+            var actualOutputLine1 = _output.GetWriteLine(0);
+            var actualOutputLine2 = _output.GetWriteLine(1);
+
+            Assert.Equal(validInput, result);
+            Assert.Equal(expectedOutputLine1, actualOutputLine1);
+            Assert.Equal(expectedOutputLine2, actualOutputLine2);
+        }
+
 
         [Theory]
-        [InlineData("0,0", 3, 3)]
-        [InlineData("0,0", 4, 3)]
-        [InlineData("2,1", 10, 5)]
-        [InlineData("2,1", 4, 6)]
-        public void ReturnLocationString_GivenValidInput(string input, int gridWidth, int gridLength)
+        [InlineData("0,0", "3,3", 3, 3)]
+        [InlineData("0,0", "4,3", 4, 3)]
+        [InlineData("2,1", "10,5", 10, 5)]
+        [InlineData("2,1", "4,6", 4, 6)]
+        public void ReturnLocationString_GivenValidInput(string input, string universeInput, int gridWidth, int gridLength)
         {
-            var universe = new Universe(gridWidth, gridLength);
-            var controller = new GameController(universe, _output, _input);
+            InputParser.ParseUniverse(universeInput);
             _input.GetReadLine(input);
             var expected = input;
 
-            var result = controller.CreateValidLocationString();
+            var result = _controller.CreateValidLocationString(gridWidth, gridLength);
 
             Assert.Equal(expected, result);
         }
 
-        //TODO: how to test error message in CreateValidLocationString
+        [Theory]
+        [InlineData(new string[] { "0,3" }, "2,2", "3,3", 3, 3)]
+        [InlineData(new string[] { "3,0" }, "0,0", "3,3" , 3, 3)]
+        [InlineData(new string[] { ",,", ";;" }, "2,2", "3,3", 3, 3)]
+        [InlineData(new string[] { "3,3,", "3,4;" }, "3,2", "4,3", 4, 3)]
+        [InlineData(new string[] { "3,3,3" },"3,2", "4,3", 4, 3)]
+        [InlineData(new string[] { "a,b" }, "1,1", "3,3", 3, 3)]
+        [InlineData(new string[] { "-1,4" }, "2,0", "3,3", 3, 3)]
+        [InlineData(new string[] { "4,-1" }, "2,2", "3,3", 3, 3)]
+        //[InlineData(new string[] { "4 4" }, "2,1", "4 3", 4, 3)] // ThrowOverflowOrFormatException

```

Comment:
Remember that the `CreateValidLocationString` function won't return until a valid input is read in. Any error you're getting with this test is because the `StubInput` has run dequeued all of the inputs that you preloaded.